src/Abstractions/NexusMods.Abstractions.GameLocators/WineParser.cs

```diff
@@ -0,0 +1,174 @@
+
+        return result;
+    }
+
+    /// <summary>
+    /// Parses the <see cref="EnvironmentVariableName"/>.
+    /// </summary>
+    public static IReadOnlyList<WineDllOverride> ParseEnvironmentVariable(ReadOnlySpan<char> environmentVariableValue)
+    {
+        if (environmentVariableValue.Length == 0) return [];
+
+        var results = new List<WineDllOverride>();
+
+        // https://gitlab.winehq.org/wine/wine/-/wikis/Wine-User's-Guide#winedlloverrides-dll-overrides
+        var span = GetWineDllOverridesSection(environmentVariableValue);
+
+        // NOTE(erri120): DLLs are separated with a semicolon
+        var splitEnumerator = span.Split(';');
+        foreach (var splitRange in splitEnumerator)
+        {
+            var section = span[splitRange];
+
+            var index = section.LastIndexOf('=');
+            if (index == -1) continue;
+
+            var namesSpan = section[..index];
+
+            var dllNamesEnumerator = namesSpan.Split(',');
+            var dllOverrideTypes = GetOverrideTypes(section);
+
+            foreach (var range in dllNamesEnumerator)
+            {
+                var name = FixDllName(namesSpan[range]).ToString();
+                results.Add(new WineDllOverride(name, dllOverrideTypes));
+            }
+        }
+
+        return results;
+    }
+
+    private static ReadOnlySpan<char> FixDllName(ReadOnlySpan<char> input)
+    {
+        return input.TrimEnd(".dll");
+    }
+
+    private static WineDllOverrideType[] GetOverrideTypes(ReadOnlySpan<char> section)
+    {
+        var index = section.LastIndexOf('=');
+        if (index == section.Length - 1) return WineDllOverride.Disabled;
+
+        var typesSpan = section[(index + 1)..];
+
+        var numTypes = typesSpan.Count(',') + 1;
+        Span<WineDllOverrideType> overrideTypesSpan = stackalloc WineDllOverrideType[numTypes];
+
+        var typesIndex = 0;
+        var typesEnumerator = typesSpan.Split(',');
+        foreach (var splitRange in typesEnumerator)
+        {
+            var typeSpan = typesSpan[splitRange];
+            if (typeSpan.Length != 1) continue;
+
+            var c = typeSpan[0];
+            if (c is 'n') overrideTypesSpan[typesIndex++] = WineDllOverrideType.Native;
+            if (c is 'b') overrideTypesSpan[typesIndex++] = WineDllOverrideType.BuiltIn;
+        }
+
+        return overrideTypesSpan[..typesIndex].ToArray();
+    }
+
+    private static ReadOnlySpan<char> GetWineDllOverridesSection(ReadOnlySpan<char> input)
+    {
+        const string prefix = "WINEDLLOVERRIDES=";
+
+        var span = input;
+        if (!span.StartsWith(prefix, StringComparison.OrdinalIgnoreCase)) return span;
+
+        span = span[prefix.Length..];
+        if (!span.StartsWith('"')) return span;
+
+        span = span[1..];
+        var index = span.IndexOf('"');
+        span = span[..index];
+
+        return span;
+    }
+}
+
+public record WineDllOverride(string DllName, WineDllOverrideType[] OverrideTypes)
```

Comment:
This should be a `record struct`, the item does not need boxing as it's composed of 2 reference types at current, meaning that it's cheaper to copy around as a value type.

As for the load orders, there's a fixed set of these,

- `b` Builtin
- `n` Native
- `b,n` Builtin then Native
- `n,b` Native then Builtin

This should ideally be represented as an enum, with each of the 4 orders represented as an enum variant.

`winecfg` UI for reference: 

![image](https://github.com/user-attachments/assets/9e504055-f144-4f86-9b1b-b50fd7587e03)

Not sure if I'm being too picky here or not though; but I thought it's worth pointing out, as the number of possibilities here is finite.

Do at least change to `record struct`.